addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java

```diff
@@ -208,6 +213,26 @@ public void updateState(YamahaReceiverState state) throws IOException {
         node = getNode(basicStatus, "Input/Input_Sel_Item_Info/Src_Number");
         value = node != null ? node.getTextContent() : "0";
         state.netRadioChannel = Integer.parseInt(value);
+
+        // Net Radio provides additional informaton like station name and currently played song
+        if ("NET RADIO".equals(state.input)) {
+            try {
+                Document docNet = postAndGetXmlResponse(
```

Comment:
The updateState() method should only cause one postAndGetXmlResponse call and no conditional other calls, for better testing. Please write an updateNetRadio() method that is called wherever necessary (Would be YamahaReceiverState.updateState() I think).